lib/moonshot/tools/asg_rollout.rb

```diff
@@ -65,6 +65,10 @@ def wait_for_in_service(new_instance)
 
           loop do
             instance_health = asg.instance_health(new_instance)
+            if instance_health.out_of_service?
```

Comment:
Tested this locally, does what it says on the tin:
```
[ \ ] [ 0m 40s ] Waiting for a new instance to join Auto Scaling Group...
[ ✗ ] [ 0m 37s ] Error while performing step: Waiting for i-03f606ef9face4526 to be InService...
  RuntimeError: Instance i-03f606ef9face4526 went OutOfService while waiting to join...
[ ✓ ] [ 0m 0s ] Restored MaxSize/DesiredCapacity values to normal!
```

However, I think we should just try again with the next instance that joins. If the user sees this happen 2-3 times they might give up and press C-c, which we'll handle. Thoughts?